### ISSUE TITLE
Opposite ring

### ISSUE THREAD

**Issue by benjub** [metamath/set.mm#2349]
Triggered by @avekens's result http://us.metamath.org/mpeuni/rmodislmod.html (see PR #2345) that a right module over a commutative ring "is" also a left module over that ring, I proposed in a comment to that PR to deduce this from the more general statement
> If M is a right R-module, then it is a left R^{op}-module.

I think the easiest way to do it would be to first define the opposite of a magma (and prove that it is a magma, and is equal to it iff the magma is commutative), then define the opposite of a ring (or actually a weak ring-like structure, like a "rg" like (NN,+,*)), using mulGrp, and similarly proving that it is a ring, equal to it iff the ring is commutative.  And finally, prove the above.  "If time permits", one could also prove the involutivity of "opposite" and the associated isomorphisms of categories.  One could do the same thing with the opposite of categories (it really is the same thing as for magmas... one should define magmoids), etc.

**Comment by jkingdon** [comment]
I'm not quite up to speed on what is being proposed here. I assume it has some connection with https://us.metamath.org/mpeuni/df-oppr.html but maybe more general in some way?

**Comment by benjub** [comment]
Yes, https://us.metamath.org/mpeuni/df-oppr.html defines the opposite ring, thank you for spotting it. So, the generalization of https://us.metamath.org/mpeuni/rmodislmod.html I mention above would use df-oppr in its statement. 

Note that df-oppr covers the case of the opposite of a ring (specifically, it is the operation in the slot for ring multiplication which is "opposited"), and not the opposite of a group-like structure or a category, for instance.
